src/credential.rs

```diff
@@ -0,0 +1,61 @@
+
+#![allow(unused_variables)]
+#![allow(dead_code)]
+
+type CredentialSubject = serde_json::Value;
+type CredentialTypes = [VerifiableCredential];
+
+
+// #[derive(Deserialize, Debug)]
+pub struct VerifiableCredential {
+	context: String,
+	id: String,
+    cred_type: String,
+	issuer: String,
+	issuance_date: String,
+	subject: CredentialSubject,
+	proof: *mut CredentialProof,
+}
+
+impl <'a> VerifiableCredential {
+    pub fn init(cred_type: &'a str, cred_subject: serde_json::Value) -> Self  {
```

Comment:
The `init` of `VerifiablePresentation` will have an array of the same parameters. So an array of cred types and cred subjects

Reply:
You can use this as a reference point of what the VP creation will need - https://www.w3.org/TR/vc-data-model/#presentations-0. There are some things that we can self determine (such as issuance date) but everything that we can't get ourselves will have to be passed in